packages/common/src/grpc-service.js

```diff
@@ -717,5 +733,19 @@ GrpcService.prototype.getService_ = function(protoOpts) {
   return service;
 };
 
+GrpcService.prototype.setGrpcMetadata_ = function(metadata) {
+  var cc = grpc.credentials.createFromMetadataGenerator(function(_, cb) {
+    var grpcMetadata = new grpc.Metadata();
+
+    for (var prop in metadata) {
+      grpcMetadata.add(prop, metadata[prop]);
+    }
```

Comment:
Thanks for the quick review! Could you show what that looks like? I tried to model this after an example snippet you gave here: https://github.com/GoogleCloudPlatform/google-cloud-node/issues/1346#issuecomment-240912476